homeassistant/components/xbox/base_sensor.py

```diff
@@ -44,7 +46,18 @@ def entity_picture(self) -> str:
         if not self.data:
             return None
 
-        return self.data.display_pic.replace("&mode=Padding", "")
+        # Xbox sometimes returns a domain that uses a wrong certificate which creates issues
+        # with loading the image.
+        # The correct domain is images-eds-ssl which can just be replaced
+        # to point to the correct image, with the correct domain and certificate
+        # using YARL URL lib, we can replace the domain part of a url and append
+        # only the queries we need (url and format) and omit unnecessary ones (padding)
+        original_url = URL(self.data.display_pic)
+        return str(
+            original_url.with_host("images-eds-ssl.xboxlive.com").with_query(
+                url=original_url.query["url"], format=original_url.query["format"]
```

Comment:
You should probably only replace the faulty one.

If url.host == xxx
  url = url.with_host(yyy)
return str(url)

Reply:
done!